compat.cmake

```diff
@@ -0,0 +1,17 @@
+
+# old gcc-4.8 support
+if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND
+  (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 4.8) AND
+  (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0))
+
+  if(EXISTS "${CMAKE_SOURCE_DIR}/ext/c4core/src/c4/compat/gcc-4.8.hpp")
+    # header injection, required to compile external projects (not necessary for c4core)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -include ${CMAKE_SOURCE_DIR}/ext/c4core/src/c4/compat/gcc-4.8.hpp")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -include ${CMAKE_SOURCE_DIR}/ext/c4core/src/c4/compat/gcc-4.8.hpp")
```

Comment:
so you'd prefer to have this directly on the gtest target ?

should the option toggle and compat.cmake be kept ?

Reply:
I think we just need to do it both for gtest and c4core. But the point is, not ryml; ryml should get it from c4core. Something like this:
```cmake
# c4Project.cmake
# function(c4_setup_test ....)
  target_compile_options(gtest PUBLIC -include .....)

# c4core/CMakeLists.txt
target_compile_options(c4core PUBLIC -include .....)
```
Now ryml will also be injected, and no compat.cmake is needed there.

As for the option toggle, not sure what you mean?

I'm sorry for the confusion; I realize this is a walk-back from what was already done in c4core and cmake. But it shouldn't be hard.

Reply:
the options below may still be relevant, but I'm not sure

Reply:
It's probably incomplete (not representative of all bad options combinations), and just options that will fail on old compilers (since requiring c++17 or other recent features).

really up to you, it may keep project simpler to get rid of this.